### ISSUE TITLE
Does Not Correctly Parse a '--' Ident

### ISSUE THREAD

**Issue by Alexhuszagh** [w3c/css-validator#298]
Using the web service, with the following example, we get a CSS parse error:

```css
.myButton {
  --: 0px;
}
```

The following warning is produced:

```
Parse Error --: 0px; 
```

However, according to the current CSS Level 3 syntax, ` --` is a [valid identifier start](https://drafts.csswg.org/css-syntax/#would-start-an-identifier), and all identifier starts do not need to be followed by other tokens in order to form a valid `<ident-token>`. I understand this is a fairly unusual edge-case.

Other standard conforming examples below all parse:

```css
.myButton1 {
  --background: 0px;
}

.myButton2 {
  -background: 0px;
}

.myButton3 {
  -\21: 0px;
}

.myButton4 {
  -\ : 0px;
}

.myButton5 {
  -\:: 0px;
}

.myButton6 {
    -\AF12 a123: 0px
}
```

While invalid CSS fails:

```css
.myButton1 {
  -\
: 0px;
}
```

In short, the CSS grammar for parsing `<ident-tokens>` is correct in all other examples, and fully conforms to the specifications, including escape tokens. However, `--` is not properly handled.

**Comment by ylafon** [comment]
fixed (partially) by #316
